scripts/core/sharedContext.yml

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+#
+# See YaML.md for syntax definition
+#
+--- #--------------------------------------------------------------------------
+type: header
+desc: "Intel $OneApi Level-Zero Extension for shared contexts."
+version: "1.6"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Shared Contexts Extension Name"
+version: "1.6"
+name: $X_SHARED_CONTEXTS_EXP_NAME
+value: '"$X_experimental_shared_contexts"'
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Shared Contexts Extension Version(s)"
+version: "1.6"
+name: $x_shared_contexts_exp_version_t
+etors:
+    - name: "1_0"
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Defines a shared context for a set of context handles."
+version: "1.6"
+class: $xDriver
+name: GetSharedContextExp
+details:
+    -  |
+       If a shared context has been previously created for the contexts passed in the array,
+       and the shared-context is the same for all, then that one is returned.
+    -  |
+       If none of the contexts in the array belong to a shared context, then a shared context is created
+       encompassing all the contexts in the array.
+    -  |
+       If some of the contexts in the array are not yet associated with any shared context, and the rest
+       belong to a single shared context, then all are added to the same shared context.
+    -  |
+       If some of the contexts in the array already belong to different shared contexts,
```

Comment:
This sentence contradicts with previous sentence.

Let's say there is shared context 1 with context 1
Now we try to create shared context , with context 1 and context 2, previous sentence 

"
If some of the contexts in the array are not yet associated with any shared context, and the rest
       belong to a single shared context, then all are added to the same shared context."

Tells us that context 2 will be added to shared context 1.

Now this sentence tells us that if context 1 belongs to shared context 1, then nullptr is returned.

Reply:
@MichalMrozek 
in your example, context 1 belongs to shared-context 1. That is this sentence:
> If only one context is passed, then its shared context is returned if it already exists.
       If not, then a new shared context is created and returned

Now, context 2 is added to shared context 1, which is this sentence:
>  If some of the contexts in the array are not yet associated with any shared context, and the rest
       belong to a single shared context, then all are added to the same shared context.

Now, the sentence you are referring to says:
> If some of the contexts in the array already belong to different shared contexts,

doesn't apply here, as context 1 and context 2 belong to the same shared-context. This sentence would apply if there's a context 3 belonging to another shared-context, and then the following call is made

GetSharedContextExp({context1, context2, context3}) 

in this case, then as the sentence says, a null shared pointer would be returned, as all them already belong to different shared contexts.

Reply:
@MichalMrozek  maybe we could clarify saying something like:

"If all contexts in the array already belong to shared-contexts, and those shared-context are not the same, then nullptr is returned"

does that sound more accurate?